src/app/shared/services/storage/storage.service.ts

```diff
@@ -11,6 +12,14 @@ export class StorageService {
     this.session = new BaseStorage(window.sessionStorage);
     this.local = new BaseStorage(window.localStorage);
   }
+
+  getStorageByType({ key, condition }: { key?: string; condition?: boolean }) {
+    const isLocal = this.local.get(key) || condition ? true : false;
```

Comment:
I think we can get rid of the ternary statement here:
```suggestion
    const isLocal = this.local.get(key) || condition;
```

Although you might have to do this to ensure type safety:
```ts
const isLocal = !!(this.local.get(key) || condition)
```

Additionally, I'm not sure about this function. I feel like it can potentially be a source of confusion since it's meant to *get* a value, but adding a `condition` value to the parameter object essentially has the effect of *setting* which storage is being read and written to in additional calls. Essentially this function has the ability to do multiple things, but is only communicating one of those things (that it gets storage by type). Ideally the calls to set which storage is being used are different from the calls to get the preferred storage type.

Initially my idea was just to keep the preferred storage type and method on AccountService, since this functionality is only for the "keepLoggedIn" behavior right now. I think it can stay on StorageService. But I'd like to have a method to set the preferred storage location that is separate from the get method.

Reply:
@meisekimiu thank you for explaining it again! I will change it tomorrow. Your suggestion is cleaner and easier to grasp.

Reply:
@meisekimiu just to clear my confusion further, I will clear the new function that I have created and add a new one to handle the storage mainly for the AccountService, is that correct?

Reply:
You can handle it in AccountService, but if you think it makes more sense in the StorageService then you can keep it in there.